src/algebra/char_zero.lean

```diff
@@ -108,10 +108,58 @@ end
 section
 variables {R : Type*} [semiring R] [no_zero_divisors R] [char_zero R]
 
+@[simp]
 lemma add_self_eq_zero {a : R} : a + a = 0 ↔ a = 0 :=
 by simp only [(two_mul a).symm, mul_eq_zero, two_ne_zero', false_or]
 
+@[simp]
 lemma bit0_eq_zero {a : R} : bit0 a = 0 ↔ a = 0 := add_self_eq_zero
+@[simp]
+lemma zero_eq_bit0 {a : R} : 0 = bit0 a ↔ a = 0 :=
+by { rw [eq_comm], exact bit0_eq_zero }
+end
+
+section
+variables {R : Type*} [ring R] [no_zero_divisors R] [char_zero R]
+
+lemma nat_mul_inj {n : ℕ} {a b : R} (h : (n : R) * a = (n : R) * b) : n = 0 ∨ a = b :=
+begin
+  rw [←sub_eq_zero, ←mul_sub, mul_eq_zero, sub_eq_zero] at h,
+  exact_mod_cast h,
+end
+
+lemma nat_mul_inj' {n : ℕ} {a b : R} (h : (n : R) * a = (n : R) * b) (w : n ≠ 0) : a = b :=
+by simpa [w] using nat_mul_inj h
+
+lemma bit0_inj {a b : R} (h : bit0 a = bit0 b) : a = b :=
+begin
+  dsimp [bit0] at h,
+  simp only [(two_mul a).symm, (two_mul b).symm] at h,
+  refine nat_mul_inj' _ two_ne_zero,
+  exact_mod_cast h,
+end
+
+lemma bit1_inj {a b : R} (h : bit1 a = bit1 b) : a = b :=
+begin
+  simp only [bit1, add_left_inj] at h,
+  exact bit0_inj h,
+end
```

Comment:
```suggestion
lemma bit1_injective : function.injective (bit1 : R → R) :=
λ a b h, begin
  simp only [bit1, add_left_inj] at h,
  exact bit0_inj h,
end
```
since according to the mathlib naming conventions, `bit1_inj` should refer to what you call `bit1_eq_bit1`.

Reply:
Done.